### ISSUE TITLE
Why are the results of referencing pre-trained teed in the model inconsistent with those of directly using teed

### ISSUE THREAD

**Issue by Etiennejiang** [xavysp/TEED#30]
teed = TED()
teed.load_state_dict(torch.load("/5_model.pth"), strict=False) 
teed.eval()  # 切换到推理模式
for param in teed.parameters():
    param.requires_grad = False

        with torch.no_grad():
            orig_sizes = [(img.shape[-2], img.shape[-1]) for img in R]
            edge_map = self.teed(R)[-1] 
        visualize_edge_map(
            edge_map=edge_map,
            output_dir="/edge",
            file_names=[f"img_test{i}.png" for i in range(len(R))],
            orig_sizes=orig_sizes
        )
def visualize_edge_map(edge_map, output_dir, file_names, orig_sizes):
    os.makedirs(output_dir, exist_ok=True)
    prob_map = torch.sigmoid(edge_map).cpu().detach().numpy()

    for i, (file_name, orig_size) in enumerate(zip(file_names, orig_sizes)):
        single_map = prob_map[i, 0]  # (H, W)
        vis_image = (255 * single_map).astype(np.uint8)
        if vis_image.shape != tuple(orig_size):
            vis_image = cv2.resize(vis_image,
                                   (orig_size[1], orig_size[0]),  # (width, height)
                                   interpolation=cv2.INTER_LINEAR)
        output_path = os.path.join(output_dir, file_name)
        cv2.imwrite(output_path, vis_image)
        print(f"Saved edge map: {output_path}")

![Image](https://github.com/user-attachments/assets/9b95b2e6-3630-4bef-80be-2bcecc082374)

![Image](https://github.com/user-attachments/assets/85f3d0c6-43fa-4f7e-aba6-42be17f35d8f)

**Comment by Etiennejiang** [comment]
R:  tensor([[[[-14.0070, -17.0070, -16.0070,  ...,  -7.0070,  -7.0070,  -7.0070],
          [-19.0070, -25.0070, -20.0070,  ...,   3.9930,   1.9930,  -4.0070],
          [-22.0070, -27.0070, -23.0070,  ...,   3.9930,   0.9930,  -8.0070],
          ...,
          [-58.0070, -57.0070, -63.0070,  ...,  37.9930,  36.9930,  33.9930],
          [-19.0070, -17.0070, -19.0070,  ...,  31.9930,  34.9930,  31.9930],
          [-15.0070, -12.0070, -10.0070,  ...,  29.9930,  31.9930,  29.9930]],

         [[-22.6690, -22.6690, -21.6690,  ...,   5.3310,   4.3310,   4.3310],
          [-19.6690, -21.6690, -19.6690,  ...,   6.3310,   5.3310,   3.3310],
          [-24.6690, -22.6690, -20.6690,  ...,   7.3310,   6.3310,   2.3310],
          ...,
          [-59.6690, -57.6690, -54.6690,  ...,  27.3310,  27.3310,  28.3310],
          [ -3.6690,   0.3310,   3.3310,  ...,  22.3310,  26.3310,  27.3310],
          [ -8.6690,  -4.6690,  -1.6690,  ...,  23.3310,  25.3310,  26.3310]],

         [[-31.6790, -31.6790, -31.6790,  ...,  -3.6790,  -4.6790,  -1.6790],
          [-27.6790, -29.6790, -28.6790,  ...,  -1.6790,  -1.6790,  -1.6790],
          [-31.6790, -30.6790, -29.6790,  ...,  -2.6790,  -0.6790,  -1.6790],
          ...,
          [-68.6790, -66.6790, -64.6790,  ...,  22.3210,  22.3210,  23.3210],
          [ -3.6790,   0.3210,   2.3210,  ...,  17.3210,  20.3210,  21.3210],
          [ -4.6790,  -0.6790,   1.3210,  ...,  16.3210,  18.3210,  18.3210]]]],
       device='cuda:0')
edge_map.shape:  torch.Size([1, 1, 228, 304])
edge_map:  tensor([[[[-0.0701, -0.0909, -0.0957,  ..., -0.1034, -0.1056, -0.0140],
          [-0.0028, -0.0330, -0.0398,  ..., -0.0510, -0.0585,  0.0292],
          [-0.0050, -0.0408, -0.0479,  ..., -0.0519, -0.0589,  0.0304],
          ...,
          [-0.0008, -0.0222, -0.0612,  ..., -0.1563, -0.0226,  0.0347],
          [-0.0018, -0.0302, -0.0660,  ..., -0.1572, -0.0253,  0.0310],
          [ 0.0180, -0.0210, -0.0394,  ..., -0.0587,  0.0308,  0.0244]]]],
       device='cuda:0')

images]:  tensor([[[[-14.0070, -17.0070, -16.0070,  ...,  -7.0070,  -7.0070,  -7.0070],
          [-19.0070, -25.0070, -20.0070,  ...,   3.9930,   1.9930,  -4.0070],
          [-22.0070, -27.0070, -23.0070,  ...,   3.9930,   0.9930,  -8.0070],
          ...,
          [-59.0070, -57.0070, -63.0070,  ...,  37.9930,  36.9930,  33.9930],
          [-19.0070, -17.0070, -19.0070,  ...,  31.9930,  34.9930,  31.9930],
          [-15.0070, -12.0070, -10.0070,  ...,  29.9930,  31.9930,  29.9930]],

         [[-22.6690, -22.6690, -21.6690,  ...,   5.3310,   4.3310,   4.3310],
          [-19.6690, -21.6690, -19.6690,  ...,   6.3310,   5.3310,   3.3310],
          [-24.6690, -22.6690, -20.6690,  ...,   7.3310,   6.3310,   2.3310],
          ...,
          [-59.6690, -57.6690, -54.6690,  ...,  27.3310,  27.3310,  28.3310],
          [ -3.6690,   0.3310,   3.3310,  ...,  22.3310,  26.3310,  27.3310],
          [ -8.6690,  -4.6690,  -1.6690,  ...,  23.3310,  25.3310,  26.3310]],

         [[-31.6790, -31.6790, -31.6790,  ...,  -3.6790,  -4.6790,  -1.6790],
          [-27.6790, -29.6790, -28.6790,  ...,  -1.6790,  -1.6790,  -1.6790],
          [-31.6790, -30.6790, -29.6790,  ...,  -2.6790,  -0.6790,  -1.6790],
          ...,
          [-69.6790, -67.6790, -65.6790,  ...,  22.3210,  22.3210,  23.3210],
          [ -3.6790,   0.3210,   2.3210,  ...,  17.3210,  20.3210,  21.3210],
          [ -4.6790,  -0.6790,   1.3210,  ...,  16.3210,  18.3210,  18.3210]]]],
       device='cuda:0')
(preds[-1].shape:  torch.Size([1, 1, 228, 304])
preds[-1]:  tensor([[[[-0.2425, -0.2486, -0.2490,  ..., -0.2500, -0.2493, -0.2426],
          [-0.2342, -0.2288, -0.2499,  ..., -0.2335, -0.2498, -0.2405],
          [-0.2130, -0.2210, -0.2459,  ..., -0.2408, -0.2425, -0.2472],
          ...,
          [ 2.5154,  4.7506,  4.8804,  ..., -0.2314, -0.2260, -0.2494],
          [ 0.2512,  1.7685,  1.9882,  ..., -0.2275, -0.2307, -0.2493],
          [-0.2404, -0.2460, -0.2112,  ..., -0.2498, -0.2494, -0.2457]]]],
       device='cuda:0')
Then it can be seen that the tensors input into the model are the same. Do I have any problems when referring to the model?
